### PR TITLE
Mark SetEventDispatcherPass as deprecated

### LINKED ISSUE #283: Deprecate ControllerAbstract::setEventDispatcher()
And `SetEventDispatcherPass`

### PR DESCRIPTION
**Pull request by szymach**
Resolves #283

### CODE REVIEW

File: Controller/ControllerAbstract.php [hunk 1]
```diff
@@ -58,6 +58,8 @@ public function __construct(
     }
 
     /**
+     * @deprecated since version 3.0 - inject dispatcher through the constructor
```
Comment (chives) [inline Controller/ControllerAbstract.php]: `since 2.1`

File: DependencyInjection/Compiler/SetEventDispatcherPass.php [hunk 2]
```diff
@@ -13,6 +13,14 @@
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 use Symfony\Component\DependencyInjection\Reference;
 
+/**
+ * @deprecated since version 3.0
```
Comment (chives) [inline DependencyInjection/Compiler/SetEventDispatcherPass.php]: `since 2.1`

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@rn0 @JarekW

**Pull request merged**